### PR TITLE
Fix non-matching bracket pairs

### PR DESCRIPTION
**Pull request by mohamadmansourX**
'Doc text' includes doc files, comments, and docstrings.

### CODE REVIEW

File: Doc/library/urllib.request.rst [hunk 1]
```diff
@@ -655,7 +655,7 @@ OpenerDirector Objects
    optional *timeout* parameter specifies a timeout in seconds for blocking
    operations like the connection attempt (if not specified, the global default
    timeout setting will be used). The timeout feature actually works only for
-   HTTP, HTTPS and FTP connections).
+   (HTTP, HTTPS and FTP connections).
```
Comment (terryjreedy) [inline Doc/library/urllib.request.rst]: Instead, remove the unneeded ')' as done elsewhere.
```suggestion
   HTTP, HTTPS and FTP connections.
```

File: Lib/distutils/msvc9compiler.py [hunk 2]
```diff
@@ -672,7 +672,7 @@ def manifest_setup_ldargs(self, output_filename, build_temp, ld_args):
     def manifest_get_embed_info(self, target_desc, ld_args):
         # If a manifest should be embedded, return a tuple of
         # (manifest_filename, resource_id).  Returns None if no manifest
-        # should be embedded.  See http://bugs.python.org/issue7833 for why
+        # should be embedded.  (see http://bugs.python.org/issue7833 for why
         # we want to avoid any manifest for extension modules if we can)
```
Comment (terryjreedy) [inline Lib/distutils/msvc9compiler.py]: Change trailing ')' to '.' and leave 'See' capitalized.
```suggestion
        # should be embedded.  See http://bugs.python.org/issue7833 for why
        # we want to avoid any manifest for extension modules if we can.
```

File: Lib/idlelib/idle_test/test_query.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-"""Test query, coverage 93%).
+"""Test query, (coverage 93%).
```
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_query.py]: I have not used () in any of the other test files.
```suggestion
"""Test query, coverage 93%.
```

File: Lib/test/support/__init__.py [hunk 4]
```diff
@@ -395,7 +395,7 @@ def dec(*args, **kwargs):
 # A constant likely larger than the underlying OS socket buffer size, to make
 # writes blocking.
 # The socket buffer sizes can usually be tuned system-wide (e.g. through sysctl
-# on Linux), or on a per-socket basis (SO_SNDBUF/SO_RCVBUF). See issue #18643
+# on Linux), or on a per-socket basis (SO_SNDBUF/SO_RCVBUF). (see issue #18643
 # for a discussion of this number).
```
Comment (terryjreedy) [inline Lib/test/support/__init__.py]: See above.
```suggestion
# on Linux), or on a per-socket basis (SO_SNDBUF/SO_RCVBUF).  See issue #18643
# for a discussion of this number.
```

File: Lib/test/test_winreg.py [hunk 5]
```diff
@@ -45,7 +45,7 @@
     ("Raw Data",      b"binary\x00data",                       REG_BINARY),
     ("Big String",    "x"*(2**14-1),                           REG_SZ),
     ("Big Binary",    b"x"*(2**14),                            REG_BINARY),
-    # Two and three kanjis, meaning: "Japan" and "Japanese")
+    # Two and three kanjis, meaning: ("Japan" and "Japanese")
```
Comment (terryjreedy) [inline Lib/test/test_winreg.py]:
```suggestion
    # Two and three kanjis, meaning: "Japan" and "Japanese".
```

File: Misc/NEWS.d/next/Build/2021-09-09-16-45-26.bpo-45067.mFmY92.rst [hunk 6]
```diff
@@ -1,5 +1,5 @@
 The ncurses function extended_color_content was introduced in 2017.
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/Build/2021-09-09-16-45-26.bpo-45067.mFmY92.rst]: This and the next suggestion go together, but github will not the two changes at once to put the URL in this sentence.
```suggestion
The ncurses function extended_color_content was introduced in 2017

```

File: Misc/NEWS.d/next/Build/2021-09-09-16-45-26.bpo-45067.mFmY92.rst [hunk 7]
```diff
@@ -1,5 +1,5 @@
 The ncurses function extended_color_content was introduced in 2017.
-https://invisible-island.net/ncurses/NEWS.html#index-t20170401) The
+(https://invisible-island.net/ncurses/NEWS.html#index-t20170401) The
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/Build/2021-09-09-16-45-26.bpo-45067.mFmY92.rst]: Goes with previous suggestion.
```suggestion
(https://invisible-island.net/ncurses/NEWS.html#index-t20170401).  The

```

File: Tools/stringbench/stringbench.py [hunk 8]
```diff
@@ -730,7 +730,7 @@ def join_alphabet_list_five(STR):
     for x in _RANGE_1000:
         sep_join(s2)
 
-@bench('"A".join(["Bob"]*100))',
+@bench('"A".join(["Bob"]*100)',
```
Comment (terryjreedy) [inline Tools/stringbench/stringbench.py:733]: Line 742 below needs the same correction.

I considered whether the 'mistake' might be intentional and checked the definition of bench.  The first argument is attached to the function f as f.comment.  So the error has no possible functional use and should be corrected.

File: Modules/clinic/arraymodule.c.h [hunk 9]
```diff
@@ -359,7 +359,7 @@ PyDoc_STRVAR(array_array_frombytes__doc__,
 "frombytes($self, buffer, /)\n"
 "--\n"
 "\n"
-"Appends items from the string, interpreting it as an array of machine values, as if it had been read from a file using the fromfile() method).");
+"Appends items from the string, interpreting it as an array of machine values, as if it had been read from a file using the fromfile() method.");
```
Comment (terryjreedy) [inline Modules/clinic/arraymodule.c.h:362]: This file is automatically generated from arraymodule.c.  It should not be edited by hand?  Did you, are did you run 'make regen' (I believe, on *nix) to regenerate it?  Or are you using windows?  I ask because something about this file caused the error.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
I checked all changes and believe they are correct except as noted.

Test failure in regenerated file check, line 1040:
```
# Updating PCbuild/_freeze_module.vcxproj.filters
The Makefile was updated, you may need to re-run make.
Error in file "./Modules/arraymodule.c" on line 3125:
Checksum mismatch!
Expected: 376001addedc67ee
Computed: f130a994f98f1227
Suggested fix: remove all generated code including the end marker,
or use the '-f' option.
```
Does anyone else know more about this?

There was also a failure in the Pipelines test, which I cannot see since tests have been restarted.  Likely the same issue, but will see after tests rerun.

**Comment by terryjreedy** [comment]
You should make PRs from PR branches derived from main.  See the devguide,  That will allow you to work on more than one independent change at a time.

File: Doc/library/pathlib.rst [hunk 10]
```diff
@@ -552,7 +552,7 @@ Pure paths provide the following methods and properties:
       Traceback (most recent call last):
         File "<stdin>", line 1, in <module>
         File "pathlib.py", line 694, in relative_to
-          .format(str(self), str(formatted)))
+          .format(str(self), str(formatted))
```
Comment (serhiy-storchaka) [inline Doc/library/pathlib.rst]: It was correct. Revert this change.

File: Include/cpython/unicodeobject.h [hunk 11]
```diff
@@ -416,7 +416,7 @@ enum PyUnicode_Kind {
 
 
 /* Fast check to determine whether an object is ready. Equivalent to
-   PyUnicode_IS_COMPACT(op) || ((PyUnicodeObject*)(op))->data.any) */
+   PyUnicode_IS_COMPACT(op) || (((PyUnicodeObject*)(op))->data.any) */
```
Comment (serhiy-storchaka) [inline Include/cpython/unicodeobject.h]: I think one of the following would be better:
```suggestion
   PyUnicode_IS_COMPACT(op) || ((PyUnicodeObject*)(op))->data.any */
```
or
```suggestion
   (PyUnicode_IS_COMPACT(op) || ((PyUnicodeObject*)(op))->data.any) */
```

**Review by serhiy-storchaka — COMMENTED** [review]
:+1:

**Comment by terryjreedy** [comment]
@zooba Do you know what is going on with Azure Pipelines? 'Details' leads to
https://dev.azure.com/Python/cpython/_build/results?buildId=88167&view=results
which has no failure indication other than maybe running for 1 sec.
Clicking around a bit got me to this page
https://dev.azure.com/Python/cpython/_build/results?buildId=88168&view=results
(88168 instead of 88167) showing all skipped or passing.

File: Doc/library/pathlib.rst [hunk 12]
```diff
@@ -552,7 +552,7 @@ Pure paths provide the following methods and properties:
       Traceback (most recent call last):
         File "<stdin>", line 1, in <module>
         File "pathlib.py", line 694, in relative_to
```
Comment (terryjreedy) [inline Doc/library/pathlib.rst]: The line # is obsolete.  Could change.  In 3.10/11
```suggestion
        File "pathlib.py", line 818, in relative_to
```
was 941 in 3.9 but not worth the bother to change in backport.

File: Doc/library/pathlib.rst (hunk 10 above)
Comment (terryjreedy) [reply Doc/library/pathlib.rst]: Agreed.  Matching '(' is on previous line of pathlib.py.

File: Include/cpython/unicodeobject.h (hunk 11 above)
Comment (terryjreedy) [reply Include/cpython/unicodeobject.h:419]: I was wondering if the outer pair is needed.  I agree that either suggestion is better.

**Review by terryjreedy — CHANGES_REQUESTED** [review]

File: Doc/library/pathlib.rst (hunk 10 above)
Comment (ambv) [reply Doc/library/pathlib.rst]:
```suggestion
          .format(str(self), str(formatted)))
```
